src/components/measurements/measurementsD3.js

```diff
@@ -0,0 +1,172 @@
+const layout = {
+  leftPadding: 120,
+  rightPadding: 30,
+  topPadding: 20,
+  bottomPadding: 50,
+  subplotHeight: 100,
+  subplotPadding: 20,
+  circleRadius: 3,
+  thresholdStrokeWidth: 2,
+  thresholdStroke: "#DDD",
+  subplotFill: "#adb1b3",
+  subplotFillOpacity: "0.15"
+};
+
+const classes = {
+  xAxis: "measurementXAxis",
+  yAxis: "measurementYAxis",
+  threshold: "measurementThreshold",
+  subplot: "measurementSubplot",
+  subplotBackground: "measurementSubplotBackground",
+  rawMeasurements: "rawMeasurements"
+};
+
+const getMeasurementDOMId = (measurement) => `meaurement_${measurement.measurementId}`;
+
+/**
+ * Creates the D3 linear scale for the x-axis with the provided measurements'
+ * values as the domain and the panelWidth with hard-coded padding values as
+ * the range. Expected to be shared across all subplots.
+ * @param {number} panelWidth
+ * @param {Array<Object>} measurements
+ * @returns {function}
+ */
+export const createXScale = (panelWidth, measurements) => {
+  return (
+    scaleLinear()
+      .domain(extent(measurements, (m) => m.value))
+      .range([layout.leftPadding, panelWidth - layout.rightPadding])
+      .nice()
+  );
+};
+
+/**
+ * Creates the D3 linear scale for the y-axis of each individual subplot with
+ * the provided measurements' jitters as the domain and the hard-coded
+ * subplot height as the range.
+ * @param {Array<Object>} measurements
+ * @returns {function}
+ */
+export const createYScale = (measurements) => {
+  return (
+    scaleLinear()
+      .domain(extent(measurements, (m) => m.measurementJitter))
+      .range([layout.subplotHeight, 0])
+      .nice()
+  );
+};
+
+/**
+ * Uses D3.groups() to aggregate measurements into a nested array of groups
+ * The groups are sorted with the most number of measurements first.
+ * @param {Array<Object>} measurements
+ * @param {string} groupBy
+ * @returns {Array<Array<string, Array>>}
+ */
+export const groupMeasurements = (measurements, groupBy) => {
+  const groupedMeasurements = groups(measurements, (d) => d[groupBy]);
+  /*
+   * Using a[1] and b[1] since d3.groups returns a nested array where the
+   * first element is the groupBy field value and the second element is the
+   * array of measurements
+   */
+  return groupedMeasurements.sort((a, b) => b[1].length - a[1].length);
+};
+
+export const clearMeasurementsSVG = (ref) => {
+  select(ref)
+    .attr("height", null)
+    .selectAll("*").remove();
+};
+
+export const drawMeasurementsSVG = (ref, svgData) => {
+  const {xScale, yScale, x_axis_label, threshold, groupedMeasurements} = svgData;
+
+  // Do not draw SVG if there are no measurements
+  if (groupedMeasurements && groupedMeasurements.length === 0) return;
+
+  const svg = select(ref);
+  const svgWidth = svg.node().getBoundingClientRect().width;
+
+  // The number of groups is the number of subplots, which determines the final SVG height
+  const totalSubplotHeight = (layout.subplotHeight * groupedMeasurements.length);
```

Comment:
Hmm, I hadn't thought about this. On one hand, it would be a great way to save space within the viz. On the other hand, this also visually devalues the groups that have less measurements. Maybe that's something we want?

Reply:
(Noting here outcome from discussion last week) this suggestion is a UI/UX idea to try one day, probably in the python version which will be quicker. It's not blocking this PR.